spring-kafka/src/main/java/org/springframework/kafka/listener/AbstractMessageListenerContainer.java

```diff
@@ -44,7 +44,7 @@
  * @author Artem Bilan
  */
 public abstract class AbstractMessageListenerContainer<K, V>
-		implements MessageListenerContainer, BeanNameAware, ApplicationEventPublisherAware, SmartLifecycle {
+		implements GenericMessageListenerContainer<K, V>, BeanNameAware, ApplicationEventPublisherAware {
```

Comment:
What happened to `SmartLifecycle` here?

Hardly ever possible, but still breaking change.

Reply:
The super.super interface (MLC) extends SL. So it was always redundant here.